packages/react-components/src/contexts/SettingsContext.tsx

```diff
@@ -87,29 +89,18 @@ export const SettingsProvider = ({
       window.location.port === '4173' || window.location.port === '5173'
     const isDev = isLocalhost && isHttp && isVite
 
-    let runnerEndpoint = new URL(window.location.href)
-    // Overwrite runner if webApp.runner is provided
-    if (webApp?.runner) {
-      runnerEndpoint = new URL(webApp.runner)
-    }
-
     const baseSettings: Settings = {
-      requireAuth: false,
-      agentEndpoint: isDev ? 'http://localhost:8080' : window.location.origin,
+      requireAuth: requireAuth ?? false,
+      agentEndpoint: agentEndpoint ?? isDev ? 'http://localhost:8080' : window.location.origin,
       webApp: {
-        runner: isDev
+        runner: webApp?.runner ?? isDev
```

Comment:
I think it makes sense for the passed in value to take precedence over any defaults – even in dev.